easybuild/tools/github.py

```diff
@@ -395,6 +407,235 @@ def post_comment_in_issue(issue, txt, repo=GITHUB_EASYCONFIGS_REPO, github_user=
+                    file_info['new'].append(not os.path.exists(full_target_path))
+
+                    mkdir(os.path.dirname(full_target_path), parents=True)
+                    shutil.copy2(path, full_target_path)
+                    _log.info("%s copied to %s", path, full_target_path)
+                except OSError as err:
+                    raise EasyBuildError("Failed to copy %s to %s: %s", path, target_path, err)
+
+                file_info['paths_in_repo'].append(target_path)
+            else:
+                raise EasyBuildError("Multiple EasyConfig instances obtained from easyconfig file %s", path)
+    else:
+        raise EasyBuildError("Subdirectory %s not found in %s", subdir, target_dir)
+
+    return file_info
+
+
+@only_if_module_is_available('git', pkgname='GitPython')
+def _easyconfigs_pr_common(paths, start_branch=None, pr_branch=None, target_account=None, commit_msg=None):
+    """
+    Common code for new_pr and update_pr functions
+
+    @paths: list of paths that will be used to create/update PR
+    @start_branch: name of branch to start from
+    @pr_branch: name of branch to push to GitHub
+    """
+    # we need files to create the PR with
+    if paths:
+        non_existing_paths = []
+        for path in paths:
+            if not os.path.exists(path):
+                non_existing_paths.append(path)
+
+        if non_existing_paths:
+            raise EasyBuildError("One or more non-existing paths specified: %s", ', '.join(non_existing_paths))
+    else:
+        raise EasyBuildError("No paths specified")
+
+    # create working directory
+    git_working_dir = tempfile.mkdtemp(prefix='git-working-dir')
+    git_repo = git.Repo.init(git_working_dir)
+    _log.debug("git working directory initialized at %s", git_working_dir)
+
+    # add remote to pull from
+    github_target_repo = build_option('github_target_repo')
+    github_url = 'git@github.com:%s/%s.git' % (target_account, github_target_repo)
+    _log.debug("Cloning from %s", github_url)
+
+    origin = git_repo.create_remote('origin', github_url)
+    if not origin.exists():
+        raise EasyBuildError("%s does not exist?", github_url)
+
+    # git fetch
+    if start_branch is None:
+        start_branch = build_option('github_target_branch')
+    _log.debug("Fetching branch %s for remote %s", start_branch, origin)
+    origin.fetch(start_branch, depth=1)
+
+    # git checkout -b <branch>; git pull
+    if hasattr(origin.refs, start_branch):
+        origin_branch = getattr(origin.refs, start_branch)
+    else:
+        raise EasyBuildError("Branch '%s' not found at %s", start_branch, github_url)
+
+    _log.debug("Checking out branch %s from remote %s", start_branch, github_url)
+    git_repo.create_head(start_branch, origin_branch).checkout()
+    origin.pull(start_branch)
+
+    # copy files to right place
+    file_info = _copy_easyconfigs_to_repo(paths, git_working_dir)
+
+    # create target branch, if it was specified
```

Comment:
Ah, hold on. I can't.

I'm using information returned by the function that copies the easyconfigs to determine the (default) name of the PR branch, i.e. the software name/version of the 1st easyconfig.

So, leaving as is.